controllers/akodeploymentconfig/cluster/cluster_controller.go

```diff
@@ -62,15 +69,24 @@ func (r *ClusterReconciler) ReconcileDelete(
 	cluster *clusterv1.Cluster,
 	_ *akoov1alpha1.AKODeploymentConfig,
 ) (ctrl.Result, error) {
-	res := ctrl.Result{}
 
 	if ctrlutil.ContainsFinalizer(cluster, akoov1alpha1.ClusterFinalizer) {
 		log.Info("Handling deleted Cluster")
+		akoPresent, err := akoInstalled() //TODO (adduarte) rename akoPresent to something more generic which represents weather or not the ako package is installed, or having problems, or crashed.
+		if err != nil {
+			log.Error(err, "could not verify if ako pkgi is present in cluster")
+			return ctrl.Result{}, err
+		}
+		if !akoPresent {
+			log.Info("The AKO package is not present in the cluster. Removing finalizer")
+			ctrlutil.RemoveFinalizer(cluster, akoov1alpha1.ClusterFinalizer)
+			return ctrl.Result{}, nil
```

Comment:
should we return right away, or should we continue below and attempt cleanup and remove finalizer even if clean up fails?

Reply:
If there is no ako installed, we don't need to execute cleanup, because that function is trying to ask AKO in the workload cluster to clean up AVI resources.